src/wallet/rpcwallet.cpp

```diff
@@ -4122,6 +4123,153 @@ UniValue walletcreatefundedpsbt(const JSONRPCRequest& request)
+                    {"changeAddress", RPCArg::Type::STR_HEX, /* default */ "pool address", "The bitcoin address to receive the change"},
+                    {"changePosition", RPCArg::Type::NUM, /* default */ "random", "The index of the change output"},
+                    {"change_type", RPCArg::Type::STR, /* default */ "set by -changetype", "The output type to use. Only valid if changeAddress is not specified. Options are \"legacy\", \"p2sh-segwit\", and \"bech32\"."},
+                    {"includeWatching", RPCArg::Type::BOOL, /* default */ "false", "Also select inputs which are watch only"},
+                    {"locktime", RPCArg::Type::NUM, /* default */ "0", "Raw locktime. Non-0 value also locktime-activates inputs"},
+                    {"lockUnspents", RPCArg::Type::BOOL, /* default */ "false", "Lock selected unspent outputs"},
+                    {"feeRate", RPCArg::Type::AMOUNT, /* default */ "not set: makes wallet determine the fee", "Set a specific fee rate in " + CURRENCY_UNIT + "/kB"},
+                    {"subtractFeeFromOutputs", RPCArg::Type::ARR, /* default */ "empty array", "A json array of integers.\n"
+                    "                              The fee will be equally deducted from the amount of each specified output.\n"
+                    "                              Those recipients will receive less bitcoins than you enter in their corresponding amount field.\n"
+                    "                              If no outputs are specified here, the sender pays the fee.",
+                        {
+                            {"vout_index", RPCArg::Type::NUM, RPCArg::Optional::OMITTED, "The zero-based output index, before a change output is added."},
+                        },
+                    },
+                    {"replaceable", RPCArg::Type::BOOL, /* default */ "false", "Marks this transaction as BIP125 replaceable.\n"
+                    "                              Allows this transaction to be replaced by a transaction with higher fees"},
+                    {"conf_target", RPCArg::Type::NUM, /* default */ "Fallback to wallet's confirmation target", "Confirmation target (in blocks)"},
+                    {"estimate_mode", RPCArg::Type::STR, /* default */ "UNSET", "The fee estimate mode, must be one of:\n"
+                    "         \"UNSET\"\n"
+                    "         \"ECONOMICAL\"\n"
+                    "         \"CONSERVATIVE\""},
+                },
+                "options"},
+        },
+        RPCResult{
+                    "{\n"
+                    "  \"complete\" : true|false, (boolean) If the transaction has a complete set of signatures\n"
+                    "  \"txid\" :                 (string) The transaction id for the send. Only 1 transaction is created regardless of \n"
+                    "                                      the number of addresses.\n",
```

Comment:
as noted in my other PR, I think `broadcast` is the wrong word here. It's moreso that we do not commit it to the wallet, which also broadcasts it

Reply:
I'm open to other terms, maybe `commit`? But that seems less clear.

The background motivation here is that an RPC consumer may need certainty about whether the transaction failed or succeeded. For example this should even work if the RPC connection gets cut before the call returns.

Just retrying is unsafe. So right now the only option is to check if the wallet contains transactions to the expected destination(s). With this flag, the RPC consumer gets full control and can for example commit the full transaction to some redundant database before broadcasting it.

(and maybe in future we support wallet-less spend based on just a descriptor with private keys, where there's no way to know if a previous attempt failed, since you can't rely on the local mempool for that)

Reply:
"broadcast" already has meaning to users unfortunately e.g., `walletbroadcast` startup arg

Reply:
That's a rather scary startup option imo; forget it once and everything gets broadcast?

Reply:
> That's a rather scary startup option imo; forget it once and everything gets broadcast?

This is definitely an option, like the proxy configuration, one'd want to have in the `bitcoin.conf` and not on the command line!

But also, not necessarily: the idea is that when you have it enabled, you broadcast wallet transactions manually (e.g. by pasting them into the transaction broadcast of a  block explorer, or a script such as `bitcoin-submittx` over tor) so they end up in the mempool or block chain through some other route. There's no reason for the wallet to broadcast them then.

Reply:
The option I introduce here doesn't even add the transaction to the wallet; it only returns the hex. So it goes beyond `walletbroadcast=0` afaik.

Reply:
I renamed `broadcast` to `add_to_wallet` and explained it a bit better in the docs.